connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
Ok, so it seems it doesn't work on older devices :(

In that case, I suggest we resort to your smart solution for lower API levels, as a kind of legacy nobody will care about in the future, but keep the solution based on isCellularDataNetworkConnected if the API level is higher.